dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgPanel.vue

```diff
@@ -186,6 +198,71 @@ export default defineComponent({
         : getCountryNameFromCountryCode(kpiValue as string) ?? kpiValue;
     },
 
+    /**
+     * Generates a list of readible strings (or just a single one) combining suppliers and their associated countries
+     * @param numberOfSuppliersPerCountryCode the map of number of suppliers and associated companies
+     * from which strings are written
+     * @returns the constructed collection of readable strings
+     */
+    generateReadableCombinationOfNumberOfSuppliersAndCountries(
+      numberOfSuppliersPerCountryCode?: Map<string, number | undefined | null>
+    ) {
+      if (numberOfSuppliersPerCountryCode != undefined) {
+        const readableListOfSuppliersAndCountries = Array.from(numberOfSuppliersPerCountryCode.entries()).map(
+          ([countryCode, numberOfSuppliers]) => {
+            const countryName = getCountryNameFromCountryCode(countryCode) ?? countryCode;
+            if (numberOfSuppliers != undefined) {
+              return String(numberOfSuppliers) + " suppliers from " + countryName;
+            } else {
+              return "There are suppliers from " + countryName;
+            }
+          }
+        );
+        if (readableListOfSuppliersAndCountries.length > 1) {
+          return readableListOfSuppliersAndCountries;
+        } else {
+          return readableListOfSuppliersAndCountries[0];
+        }
+      } else {
+        return null;
+      }
+    },
+
+    /**
+     * Converts the map of ProcurementCategory and LksgProductCategory into an array for a proper handling of the
+     * DetailsCompanyDataTable in the LksgCompanyDataTable (modal showing information related to Procurement Categories)
+     * @param inputObject Map to convert to array
+     * @returns The constructed map
+     */
+    reformatProcurementCategoriesValue(inputObject: Map<ProcurementCategoryType, LksgProcurementCategory> | null) {
+      if (inputObject == null) return null;
+      const listOfProductCategories = [];
+      for (const [procurementCategory, lksgProductCategory] of Object.entries(inputObject)) {
+        const definitionsOfProductTypeOrService = function (): string[] | string {
+          if ((lksgProductCategory as LksgProcurementCategory).procuredProductTypesAndServicesNaceCodes?.length > 1) {
+            return (lksgProductCategory as LksgProcurementCategory).procuredProductTypesAndServicesNaceCodes;
+          } else {
+            return (lksgProductCategory as LksgProcurementCategory).procuredProductTypesAndServicesNaceCodes[0];
```

Comment:
This could lead to an index-out-of-bounds exception if the list is empty, right?   Or is that case impossible?

Reply:
Ok I just found out that in JavaScript `undefined` is returned if you access an index which does not exist.  So this would return `undefined`.   IS this intended by you?

Reply:
Yes I think we need a safe guard here and I've seen that you have already implemented it.

In the other conversation, I suggested to use the `??` operator here.  We can discuss it there, so I can close this conversation here.